enhancements/windows-containers/csi-proxy.md

```diff
@@ -0,0 +1,207 @@
+
+### Workflow Description
+
+1. The cluster administrator will identify what CSI storage drivers the cluster requires for its workloads.
+2. Based on the requirements of the workloads, the cluster admin can follow upstream documentation to deploy the CSI
+   node drivers on its Windows nodes.
+
+### API Extensions
+
+N/A
+
+### Risks and Mitigations
+
+* If there are any security issues present in the CSI driver images deployed by the user, they will not be fixed
+  by upgrading OCP. It is up to the user to stay informed and update their images as necessary.
+* A mismatch between the OpenShift Linux CSI drivers and the user deployed Windows CSI drivers could cause undefined
+  behavior. This is described in the [version skew](#version-skew-strategy) section.
+
+### Drawbacks
+
+The main drawback to this approach is it puts the onus on the cluster administrator. As they will be required to deploy
+and maintain the Windows CSI node driver DaemonSet themselves, they will only be given support for issues with either
+the CSI driver controller, or the csi-proxy Windows service. Support will not be given for issues originating within
+the upstream CSI node drivers.
+
+This may cause an increased load on both support and developers, as support issues coming in will need to be
+properly filtered out based on the root cause of the problem.
+
+## Design Details
+
+### Test Plan
+
+WMCO already has an extensive end to end test suite. To facilitate testing the additions described in this enhancement,
+the e2e tests will be augmented to deploy a CSI node driver DaemonSet across Windows nodes, just as a user would. The
+e2e tests would then validate that a Windows pod is able to mount a volume backed by the driver. Initially vSphere
+storage will be tested, and other platforms can be tested in the future as desired. The purpose of testing is
+to ensure CSI proxy is properly interacting with the Windows instance and Windows pods.
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+N/A
+
+#### Tech Preview -> GA
+
+This feature is targeted for WMCO 9.0.0 in OpenShift 4.14.
+
+#### Removing a deprecated feature
+
+No action required.
+
+### Upgrade / Downgrade Strategy
+
+#### Cluster upgrades
+
+[When upgrading cluster versions, in-tree driver Volumes will automatically be updated to use the appropriate CSI node
+drivers by the cluster](https://docs.openshift.com/container-platform/4.12/storage/container_storage_interface/persistent-storage-csi-migration.html).
+In order to properly make use of this, and ensure a clean transition with minimal workload interruption, the CSI node
+drivers must be running on the Windows node before the upgrade commences. As the node drivers require csi-proxy to run,
+as this feature is targeted for WMCO 9.0.0, a new 8.y.z version of WMCO must be released which runs csi-proxy on each
+Windows node. If this feature is backported to WMCO x.y.z, the same must be done for WMCO x-1.y.z. This will allow the
```

Comment:
>If this feature is backported to WMCO x.y.z, the same must be done for WMCO x-1.y.z.

Is that true? This implies we have to backport CSI-proxy to all supported WMCO versions.

Reply:
I'm not sure I follow, if we decided to backport the out-of-tree storage feature to 4.12, this only means we have to backport CSI proxy to 4.11.